tests/vcs/VCSCommitBearTest.py

```diff
@@ -0,0 +1,179 @@
+
+            shutil.which = lambda *args, **kwargs: 'path/to/git'
+            self.assertTrue(GitCommitBear.check_prerequisites())
+        finally:
+            shutil.which = _shutil_which
+
+    def test_head_commit(self):
+        self.assertEqual(self.run_uut(), ['HEAD commit information'])
+
+
+@generate_skip_decorator(GitCommitBear)
+class AnalyzeGitCommitTest(unittest.TestCase):
+
+    @staticmethod
+    def run_git_command(*args, stdin=None):
+        run_shell_command(' '.join(('git',) + args), stdin)
+
+    @staticmethod
+    def git_commit(msg):
+        # Use stdin mode from git, since -m on Windows cmd does not support
+        # multiline messages.
+        AnalyzeGitCommitTest.run_git_command('commit',
+                                             '--allow-empty',
+                                             '--allow-empty-message',
+                                             '--file=-',
+                                             stdin=msg)
+
+    def run_uut(self, *args, **kwargs):
+        """
+        Runs the unit-under-test (via `self.uut.run()`) and collects the
+        messages of the yielded results as a list.
+
+        :param args:   Positional arguments to forward to the run function.
+        :param kwargs: Keyword arguments to forward to the run function.
+        :return:       A list of the message strings.
+        """
+        return list(result.message for result in self.uut.run(*args, **kwargs))
+
+    def assert_no_msgs(self):
+        """
+        Assert that there are no messages in the message queue of the bear, and
+        show the messages in the failure messgae if it is not empty.
+        """
+        self.assertTrue(
+            self.msg_queue.empty(),
+            'Expected no messages in bear message queue, but got: ' +
+            str(list(str(i) for i in self.msg_queue.queue)))
+
+    def setUp(self):
+        self.msg_queue = Queue()
+        self.section = Section('')
+        self.uut = GitCommitBear(None, self.section, self.msg_queue)
+
+        self._old_cwd = os.getcwd()
+        self.gitdir = mkdtemp()
+        os.chdir(self.gitdir)
+        self.run_git_command('init')
+        self.run_git_command('config', 'user.email coala@coala.io')
+        self.run_git_command('config', 'user.name coala')
+
+    @staticmethod
+    def _windows_rmtree_remove_readonly(func, path, excinfo):
+        os.chmod(path, stat.S_IWRITE)
+        func(path)
+
+    def tearDown(self):
+        os.chdir(self._old_cwd)
+        if platform.system() == 'Windows':
+            onerror = self._windows_rmtree_remove_readonly
+        else:
+            onerror = None
+        shutil.rmtree(self.gitdir, onerror=onerror)
+
+    def test_analyze_git_commit(self):
+        run_shell_command('echo "abc" >> testfile1.txt ')
+        run_shell_command('git add testfile1.txt')
+        run_shell_command('git commit -m "Add testfile1"')
+        self.assertEqual(self.run_uut(), [])
```

Comment:
and also, test the result by using `get_results()` (See `URLHeadBear` for the example) as we're not matching only the message here, we're also matching the contents of the `CommitResult.contents`

Reply:
See https://github.com/coala/coala-bears/pull/2543/files#r197507946